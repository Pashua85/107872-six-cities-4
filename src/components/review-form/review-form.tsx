import React, {SyntheticEvent} from 'react';
import withCommentText from '../../hocs/withCommentText';

interface IRadioButton {
  value: string,
  id: string,
  checked: boolean
}

interface ReviewFormProps {
  commentText: string,
  disabled: boolean,
  onCommentTextChange: (e: SyntheticEvent) => void,
  onRatingChange: (e: SyntheticEvent) => void,
  onFormSubmit: (e: SyntheticEvent) => void,
  radioButtons: IRadioButton[],
  errorMessage: string
  match: {
    params: {
      id: string
    }
  }
}

const ReviewForm: React.FC<ReviewFormProps> = (props) => {
  const {commentText, disabled, onCommentTextChange, onRatingChange, onFormSubmit, radioButtons, errorMessage} = props;

  return (
    <form
      className="reviews__form form"
      onSubmit={onFormSubmit}
    >
      <label className="reviews__label form__label" htmlFor="review">Your review</label>
      <div className="reviews__rating-form form__rating">
        {
          radioButtons.map((rb) => (
            <React.Fragment key={rb.id}>
              <input className="form__rating-input visually-hidden" name="rating" value={rb.value} id={rb.id} type="radio" onChange={onRatingChange} checked={rb.checked} />
              <label htmlFor={rb.id} className="reviews__rating-label form__rating-label" title="perfect">
                <svg className="form__star-image" width="37" height="33">
                  <use href="#icon-star"></use>
                </svg>
              </label>
            </React.Fragment>
          ))
        }
      </div>
      <textarea
        className="reviews__textarea form__textarea"
        id="review"
        name="review"
        placeholder="Tell how was your stay, what you like and what can be improved"
        value={commentText}
        onChange={onCommentTextChange}
      >
      </textarea>
      {
        errorMessage.length > 0 &&
        <p className="reviews__help" style={{color: `firebrick`}}>
          {errorMessage}
        </p>
      }
      <div className="reviews__button-wrapper">
        <p className="reviews__help">
          To submit review please make sure to set <span className="reviews__star">rating</span> and describe your stay with at least <b className="reviews__text-amount">50 characters</b>.
        </p>
        <button className="reviews__submit form__submit button" type="submit" disabled={disabled}>Submit</button>
      </div>
    </form>
  );
};

export default withCommentText(ReviewForm);
export {ReviewForm};


