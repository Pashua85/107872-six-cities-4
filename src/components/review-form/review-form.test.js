import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ReviewForm} from './review-form';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`ReviewForm`, () => {
  it(`should render correctly`, () => {
    const wrapper = shallow(
        <ReviewForm
          commentText=""
          disabled={true}
          onCommentTextChange={() => {}}
          onRatingChange={() => {}}
          onFormSubmit={() => {}}
          radioButtons={[]}
          errorMessage=""
        />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
