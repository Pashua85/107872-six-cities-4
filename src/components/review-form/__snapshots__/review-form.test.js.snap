// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm should render correctly 1`] = `ShallowWrapper {}`;
